api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerInstanceTypePriceTypeAndCloudRegionTest.java

```diff
@@ -0,0 +1,428 @@
+
+        assertThrows(e -> e.getMessage().contains(NOT_ALLOWED), () -> launchPipeline(configuration, INSTANCE_TYPE));
+
+        verify(instanceOfferManager).isInstanceAllowed(eq(INSTANCE_TYPE), eq(ID_2), eq(true));
+    }
+
+    @Test
+    public void launchPipelineShouldFailOnNotAllowedInstanceType() {
+        doReturn(false).when(instanceOfferManager).isInstanceAllowed(eq(INSTANCE_TYPE), eq(ID), eq(true));
+        doReturn(NOT_ALLOWED).when(messageHelper).getMessage(eq(ERROR_INSTANCE_TYPE_IS_NOT_ALLOWED), eq(INSTANCE_TYPE));
+
+        assertThrows(e -> e.getMessage().contains(NOT_ALLOWED), () -> launchPipeline(configuration, INSTANCE_TYPE));
+
+        verify(instanceOfferManager).isInstanceAllowed(eq(INSTANCE_TYPE), eq(ID), eq(true));
+    }
+
+    @Test
+    public void launchPipelineShouldNotValidatePipelineInstanceTypeIfItIsNotSpecified() {
+        launchPipeline(configuration, null);
+
+        verify(instanceOfferManager, times(0)).isInstanceAllowed(any(), eq(ID), eq(true));
+        verify(instanceOfferManager).isPriceTypeAllowed(eq(SPOT), eq(null), eq(false));
+    }
+
+    @Test
+    public void launchPipelineShouldValidatePriceTypeAsOnDemandIfItIsNotSpecified() {
+        configuration.setIsSpot(null);
+
+        launchPipeline(configuration, INSTANCE_TYPE);
+
+        verify(instanceOfferManager).isInstanceAllowed(eq(INSTANCE_TYPE), eq(ID), eq(false));
+        verify(instanceOfferManager).isPriceTypeAllowed(eq(ON_DEMAND), eq(null), eq(false));
+    }
+
+    @Test
+    public void launchPipelineShouldFailOnNotAllowedPriceType() {
+        doReturn(false).when(instanceOfferManager).isPriceTypeAllowed(eq(SPOT), eq(null), eq(false));
+        doReturn(NOT_ALLOWED).when(messageHelper).getMessage(eq(ERROR_PRICE_TYPE_IS_NOT_ALLOWED), eq(PriceType.SPOT));
+
+        final Runnable result = () -> launchPipeline(configuration, INSTANCE_TYPE);
+        assertThrows(e -> e.getMessage().contains(NOT_ALLOWED), result);
+
+        verify(instanceOfferManager).isInstanceAllowed(eq(INSTANCE_TYPE), eq(ID), eq(true));
+        verify(instanceOfferManager).isPriceTypeAllowed(eq(SPOT), eq(null), eq(false));
+    }
+
+    @Test
+    public void launchPipelineShouldFailIfToolCloudRegionIsConfiguredAndItDiffersFromRunConfigurationOne() {
+        final PipelineConfiguration toolConfiguration = new PipelineConfiguration();
+        toolConfiguration.setCloudRegionId(ID_3);
+        doReturn(toolConfiguration).when(pipelineConfigurationManager)
+                .getConfigurationForTool(eq(tool), eq(configuration));
+        doReturn(NOT_ALLOWED).when(messageHelper).getMessage(eq(ERROR_TOOL_CLOUD_REGION_NOT_ALLOWED), any(), any());
+
+        assertThrows(e -> e.getMessage().contains(NOT_ALLOWED), () -> launchPipeline(configuration, null));
+        assertThrows(e -> e.getMessage().contains(NOT_ALLOWED), () -> launchTool(configuration, null));
```

Comment:
Let's skip mocking and checking the error message. It made sense in the original test but now it doesn't.

Reply:
Done for every case where we have assertThrows.